File: v3-sandbox/prototype/sdk/src/main/java/org/hiero/sdk/simple/network/ConsensusNode.java
package org.hiero.sdk.simple.network;

import org.jspecify.annotations.NonNull;

public record ConsensusNode(@NonNull String ip, @NonNull String port, @NonNull String account) {

    /**
     * Get the address of the consensus node. The address is the IP address and port of the consensus node.
     *
     * @return the address
     */
    @NonNull
    public String getAddress() {
        return ip + ":" + port;
    }

    /**
     * Get the account ID of the consensus node.
     *
     * @return the account ID
     */
    @NonNull
    public AccountId getAccountId() {
        return AccountId.from(account);
    }
}

File: v3-sandbox/prototype-typescript/src/common/ConsensusNode.ts
import type { AccountId } from "./AccountId.js";

export class ConsensusNode {
  public readonly ip: string;
  public readonly port: number;
  public readonly account: AccountId;

  public constructor(ip: string, port: number, account: AccountId) {
    this.ip = ip;
    this.port = port;
    this.account = account;
  }
}
